govcd/access_control.go

```diff
@@ -354,3 +360,102 @@ func (adminCatalog *AdminCatalog) getAccessControlHeader(useTenantContext bool)
+// or can set access control for specific users/groups, passing isSharedToEveryOne false, everyoneAccessLevel "" and accessSettings filled as desired.
+// The method will fail if tries to configure access control for everybody and passes individual users/groups to configure.
+// It returns the control access parameters that are read from the API (using Vdc.GetControlAccess).
+func (vdc *Vdc) SetControlAccess(isSharedToEveryOne bool, everyoneAccessLevel string, accessSettings []*types.AccessSetting, useTenantContext bool) (*types.ControlAccessParams, error) {
+	err := checkSanityVdcControlAccess(vdc)
+	if err != nil {
+		return nil, err
+	}
+
+	if (isSharedToEveryOne && accessSettings != nil) && len(accessSettings) > 0 {
+		return nil, fmt.Errorf("either configure access for everybody or individual users, not both at the same time")
+	}
+
+	var tenantContextHeaders map[string]string
+	var accessControl = &types.ControlAccessParams{
+		Xmlns: types.XMLNamespaceVCloud,
+	}
+
+	if isSharedToEveryOne { // Do configuration for everyone
+		if everyoneAccessLevel == "" {
+			return nil, fmt.Errorf("everyoneAccessLevel needs to be set if isSharedToEveryOne is true")
+		}
+
+		accessControl.IsSharedToEveryone = true
+		accessControl.EveryoneAccessLevel = takeStringPointer(everyoneAccessLevel)
+
+	} else { // Do configuration for individual users/groups
+		if len(accessSettings) > 0 {
+			accessControl.AccessSettings = &types.AccessSettingList{
+				AccessSetting: accessSettings,
+			}
+		}
+	}
+
+	if useTenantContext {
+		tenantContext, err := vdc.getTenantContext()
+		if err != nil {
+			return nil, fmt.Errorf("error getting the tenant context - %s", err)
+		}
+
+		tenantContextHeaders = getTenantContextHeader(tenantContext)
+	}
+
+	err = vdc.client.setAccessControlWithHttpMethod(http.MethodPut, accessControl, vdc.Vdc.HREF, "vdc", vdc.Vdc.Name, tenantContextHeaders)
+	if err != nil {
+		return nil, fmt.Errorf("there was an error when setting VDC control access params - %s", err)
+	}
+
+	return vdc.GetControlAccess(useTenantContext)
+}
+
+// DeleteControlAccess makes stop sharing VDC with anyone
+func (vdc *Vdc) DeleteControlAccess(useTenantContext bool) (*types.ControlAccessParams, error) {
+	return vdc.SetControlAccess(false, "", nil, useTenantContext)
+}
+
+// checkSanityVdcControlAccess is a function that check some Vdc attributes and returns error if any is missing. It is useful for
+// checking sanity of Vdc struct before running controlAccess methods.
+func checkSanityVdcControlAccess(vdc *Vdc) error {
+	if vdc.client == nil {
+		return fmt.Errorf("client has not been set up on Vdc struct. Please initialize it before using this method")
+	}
+	if vdc.Vdc == nil {
+		return fmt.Errorf("types.Vdc struct has not been set up on Vdc struct. Please initialize it before using this method ")
+	}
+	if vdc.Vdc.Name == "" {
```

Comment:
though I think two checks are enough. As we need client and name and that's it.
```suggestion
	if vdc.Vdc == nil|| vdc.Vdc.Name == "" {
```

Reply:
Done!